internal/analyses.go

```diff
@@ -34,13 +33,11 @@ func (i *Internal) GetAsyncData(writer http.ResponseWriter, request *http.Reques
 
 	deployments, err := i.deploymentList(i.ViceNamespace, filter)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		return err
 	}
 
 	if len(deployments.Items) < 1 {
-		common.Error(writer, "no deployments found", http.StatusInternalServerError)
-		return
+		return errors.New("no deployments found")
```

Comment:
We'd have to verify that nothing that calls this service depends on a `500` status in this case, but it might be kind of nice to return a `404` status in this case instead. That would help to ensure that callers know that the error was most likely caused by a bad request.

Reply:
Should be fixed now.